llvm/lib/Target/AMDGPU/VOP3PInstructions.td

```diff
@@ -709,7 +708,7 @@ multiclass MAIInst<string OpName, string P, SDPatternOperator node> {
                                  !if(!eq(node, null_frag), null_frag, AgprMAIFrag<node>)>,
                          MFMATable<1, NAME # "_e64">;
 
-        let SubtargetPredicate = isGFX90APlus in
+        let OtherPredicates = [isGFX90APlus] in
```

Comment:
Looks like the outer "let SubtargetPredicate = HasMAIInsts in {" would have dominated this, so having isGFX90APlus is an additional restriction now. Is that correct? If so LGTM

Reply:
The VGPR encoded version was only added in 90a, this was always the restriction. The inner let is still a bit of a hack, but it's at least more local now